modules/currency-l0/src/main/scala/io/constellationnetwork/currency/l0/snapshot/CurrencySnapshotConsensusStateAdvancer.scala

```diff
@@ -294,7 +294,7 @@ object CurrencySnapshotConsensusStateAdvancer {
                         gossipForkInfo(gossip, signedMajorityArtifact) >>
                         maybeDataApplication.traverse_ { da =>
                           signedMajorityArtifact.toHashed >>= da.onSnapshotConsensusResult
-                        }
+                        }.handleErrorWith(logger.error(_)("Unhandled exception during onSnapshotConsensusResult"))
```

Comment:
What happens with the consensus then? I believe the consensus shouldn't progress if there is non-recoverable issue in data application. Right now the error would be "swallowed" and what then?

Reply:
The `onSnapshotConsensusResult` is just a side effecting operation and does not return a value for consensus so I thought it would be appropriate to handle in this way. The issue that caused this was that I was trying to push data to an external API in this method but occasionally the library invoking the API would throw an exception. It would seem more fail-proof to have the metagraph developer add in additional exception handling in their implementation if this method is critical to the functioning of the application, rather than the case now where a poorly designed method can completely crash the application.

Reply:
well not "crash" but it completely stalls the consensus and the only log messages that begin to appear are something like "Facilitating consensus..." one after another. This makes it unclear even what happened so that the developer could identify what portion of the application to debug.

Reply:
ok, got it :+1: if that's fire and forget, then it seems fine